mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -367,6 +371,13 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
         return Optional.empty();
     }
 
+    private void validateMailboxBelongsToUser(MailboxSession mailboxSession, MailboxPath mailboxPath) throws MailboxException {
```

Comment:
the mailbox doesn't exist yet so you can't validate anything about Mailbox.
furthermore, I would use "assert" instead of "validate".
Proposal: assertMailboxPathBelongToUser